linkerd/config/src/main/scala/io/buoyant/linkerd/config/types/ThriftProtocolDeserializer.scala

```diff
@@ -17,3 +17,18 @@ class ThriftProtocolDeserializer extends ConfigDeserializer[TProtocolFactory] {
       }
     }
 }
+
+class ThriftProtocolSerializer extends ConfigSerializer[TProtocolFactory] {
+  override def serialize(
+    value: TProtocolFactory,
+    jgen: JsonGenerator,
+    provider: SerializerProvider
+  ): Unit = {
+    val protocol = value match {
+      case _: TBinaryProtocol.Factory => "binary"
```

Comment:
Unfortunately `TBinaryProtocol.Factory` is not the same type that's produced by calling `Protocols.binaryFactory()` above, so this deserializer always returns `thriftProtocol: unknown` when `thriftProtocol: binary` is specified in a config.  How hard would it be to update ConfigHandlerTest to also test this field?